src/snoowrap.js

```diff
@@ -233,6 +233,32 @@ const snoowrap = class snoowrap {
       return requester;
     });
   }
+  static fromInstalledClientAuth ({
+    userAgent = isBrowser ? global.navigator.userAgent : requiredArg('userAgent'),
+    clientId = requiredArg('clientId'),
+    deviceId = requiredArg('deviceId'),
+    endpointDomain = 'reddit.com'
+  }) {
+    return this.prototype.credentialedClientRequest.call({
+      clientId,
+      deviceId,
+      // Use `this.prototype.rawRequest` function to allow for custom `rawRequest` method usage in subclasses.
+      rawRequest: this.prototype.rawRequest
+    }, {
+      method: 'post',
+      baseUrl: `https://www.${endpointDomain}/`,
+      uri: 'api/v1/access_token',
+      form: {grant_type: 'https://oauth.reddit.com/grants/installed_client', device_id: deviceId}
```

Comment:
Is there a reason this uses the `https://oauth.reddit.com/grants/installed_client` grant type rather than the `client_credentials` grant type? Would it be worthwhile to have an option for both?

Reply:
The difference seems a little vague. From what I've gathered

* installed_client is for mobile devices
* client_credentials is for web apps

In my case I'm looking to implement this using React Native. So I opted for installed_client for now. I can definitely add in a parameter to choose grant type and then make deviceId optional.

Reply:
Yes, I think that would be a good idea.